types/appconfig.pp

```diff
@@ -0,0 +1,24 @@
+type Puppetwebhook::Config = Struct[{
+  protected                 => Boolean,
+  user                      => String,
+  pass                      => String,
+  client_cfg                => Stdlib::Absolutepath,
+  client_timeout            => String,
+  use_mco_ruby              => Boolean,
+  use_mcollective           => Boolean,
+  discovery_timeout         => String,
+  chatops                   => Boolean,
+  Optional[chatops_service] => Enum['slack', 'rocketchat'],
+  Optional[chatops_channel] => Pattern[/\A#.*/],
+  Optional[chatops_user]    => String,
+  Optional[chatops_url]     => Variant[Stdlib::HTTPUrl, Stdlib::HTTPSUrl],
```

Comment:
This can be just `Stdlib::HTTPUrl` as that also allows https.
> [`type Stdlib::HTTPUrl = Pattern[/(?i:^https?:\/\/)/]`](https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/types/httpurl.pp)